### PR TITLE
Add support for PersonPicture, VSM Setters bindings and complex objects

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): https://github.com/windows-toolkit/Graph-Controls/issues/10

## PR Type

What kind of change does this PR introduce?
- Bugfix
- Feature

## What is the new behavior?
* [Wasm] Fix unable to reset `Image.Source` property
* Add support for `PersonPicture`
* Add support for `VisualState` `Setter` data binding, static resources and complex objects

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by jeromelaban** [comment]
Here's what it looks like: 
image

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 1]
```diff
@@ -3776,7 +3806,99 @@ private void BuildChild(IIndentedStringBuilder writer, XamlMemberDefinition owne
             }
         }
 
-        private bool HasChildrenWithPhase(XamlObjectDefinition xamlObjectDefinition)
+		//private void BuildVisualStateSetter(IIndentedStringBuilder writer, string fullTargetType, string lineEnding, string property, XamlMemberDefinition valueNode, string targetInstance = null)
```
Comment (davidjohnoliver) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs]: Commented code

File: src/Uno.UI/Helpers/WinUI/StringUtil.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Uno.UI.Helpers.WinUI
+{
+	internal static class StringUtil
```
Comment (davidjohnoliver) [inline src/Uno.UI/Helpers/WinUI/StringUtil.cs]: Maybe some documentation what/why (mainly to discourage people from using it in non-imported code)

File: src/Uno.UI/UI/Xaml/Controls/PersonPicture/PersonPicture.cs [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+		private readonly string SR_BadgeItemPlural6			= "BadgeItemPlural6";
+		private readonly string SR_BadgeItemPlural1			= "BadgeItemPlural1";
+		private readonly string SR_BadgeItemPlural3			= "BadgeItemPlural3";
+		private readonly string SR_BadgeItemPlural4			= "BadgeItemPlural4";
+
+		public PersonPictureTemplateSettings TemplateSettings { get; private set; }
+
+		public PersonPicture()
+		{
+			// DefaultStyleKey = typeof(PersonPicture);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/PersonPicture/PersonPicture.cs]: should be safe to set this, it'll just do nothing (yet)? Or am I missing something?

File: src/Uno.UI/Uno.UI.csproj [hunk 4]
```diff
@@ -347,6 +347,12 @@
 		</ProjectReference>
 	</ItemGroup>
 
+	<ItemGroup>
```
Comment (davidjohnoliver) [inline src/Uno.UI/Uno.UI.csproj]: Remove these changes

File: src/Uno.UWP/Uno.csproj [hunk 5]
```diff
@@ -38,6 +38,10 @@
 	<ItemGroup>
 		<UpToDateCheckInput Include="**\*.cs" Exclude="bin\**\*.cs;obj\**\*.cs;" />
 	</ItemGroup>
+	
+	<ItemGroup>
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Uno.csproj]: Remove

File: src/Uno.UI/UI/Xaml/Controls/PersonPicture/PersonPicture.cs (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/PersonPicture/PersonPicture.cs]: True, adding it back

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 6]
```diff
@@ -82,8 +82,9 @@ internal partial class XamlFileGenerator
         private readonly INamedTypeSymbol _contentPresenterSymbol;
         private readonly INamedTypeSymbol _stringSymbol;
         private readonly INamedTypeSymbol _objectSymbol;
-        private readonly INamedTypeSymbol _iFrameworkElementSymbol;
-        private readonly INamedTypeSymbol _dependencyObjectSymbol;
+		private readonly INamedTypeSymbol _iFrameworkElementSymbol;
+		private readonly INamedTypeSymbol _uiElementSymbol;
```
Comment (carldebilly) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs]: tab vs space?

File: src/Uno.UI.RuntimeTests/MUX/Utilities/MSTestInterop.cs [hunk 7]
```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !USING_TAEF
```
Comment (carldebilly) [inline src/Uno.UI.RuntimeTests/MUX/Utilities/MSTestInterop.cs]: `TAEF` ?

File: src/Uno.UI/UI/Xaml/Setter.cs [hunk 8]
```diff
@@ -101,7 +101,8 @@ private BindingPath TryGetOrCreateBindingPath(DependencyPropertyValuePrecedences
 
 			if (Target == null)
 			{
-				throw new InvalidOperationException($"Unable to apply setter value with null {nameof(Target)}");
+				// Ignore a null setter, act as a no-op.
+				return null;
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Setter.cs]: It's strange we never got this before...

**Pull request merged**